mazegame/Assets/Scripts/MazeRunner/MazeRunner.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MazeRunner : MonoBehaviour
+{
+
+    public float walkSpeed;
+    public float rotationSpeed;
+
+    public Transform rotationTransform;
+
+    Vector2 direction = Vector2.zero;
+
+    int targetX = 1;
+    int targetY = 1;
+
+    int currentX = 1;
+    int currentY = 1;
+
+    float currentAngle;
+    float lastAngle;
+
+    void Update()
+    {
+        bool targetReached = transform.position.x == targetX && transform.position.y == targetY;
+
+        currentX = Mathf.FloorToInt(transform.position.x);
+        currentY = Mathf.FloorToInt(transform.position.y);
+
+        direction.x = Input.GetAxisRaw("Horizontal");
+        direction.y = Input.GetAxisRaw("Vertical");
+
+        float angle = 0;
+
+        if (direction.x > 0)
+        {
+            angle = 270;
+
+            if (MazeGenerator.instance.GetMazeGridCell(currentX + 1, currentY) && targetReached)
+            {
+                targetX = currentX + 1;
+                targetY = currentY;
+            }
+        }
+        else if (direction.x < 0)
+        {
+            angle = 90;
+
+            if (MazeGenerator.instance.GetMazeGridCell(currentX - 1, currentY) && targetReached)
+            {
+                targetX = currentX - 1;
+                targetY = currentY;
+            }
+        }
+        else if (direction.y > 0)
+        {
+            angle = 0;
+
+            if (MazeGenerator.instance.GetMazeGridCell(currentX, currentY + 1) && targetReached)
+            {
+                targetX = currentX;
+                targetY = currentY + 1;
+            }
+        }
+        else if (direction.y < 0)
+        {
+            angle = 180;
+
+            if (MazeGenerator.instance.GetMazeGridCell(currentX, currentY - 1) && targetReached)
+            {
+                targetX = currentX;
+                targetY = currentY - 1;
+            }
+        }
+        else
+        {
+            angle = lastAngle;
+        }
```

Comment:
Please use a switch statement. Only one variable is being checked here

Reply:
Multiple variables, can't do switch statement